Swift/Tests/VectorSearchTest.swift

```diff
@@ -0,0 +1,400 @@
+    
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+    }
+    
+    override func initDB() throws {
+        var config = DatabaseConfiguration()
+        config.directory = self.directory
+     
+        let res = ("Support/databases/vectorsearch" as NSString).appendingPathComponent("words_db")
+        let path = Bundle(for: Swift.type(of:self)).path(forResource: res, ofType: "cblite2")
+        
+        if !Database.exists(withName: "words_db") {
+            do {
+                try Database.copy(fromPath: path!, toDatabase: databaseName, withConfig: config)
+            } catch {
+                fatalError("Couldn't load words_db")
+            }
+        }
+        try openDB()
+    }
+    
+    func toDocIDWordMap(rs: ResultSet) -> [String: String] {
+        var wordMap: [String: String] = [:]
+        for result in rs.allResults() {
+            if let docID = result.string(at: 0),
+               let word = result.string(at: 1) {
+                wordMap[docID] = word
+            }
+        }
+        return wordMap
+    }
+    
+/*
+ 1. TestVectorIndexConfigurationDefaultValue
+    Description
+        Test that the VectorIndexConfiguration has all default values returned as expected.
+    Steps
+        1. Create a VectorIndexConfiguration object.
+            - expression: "vector"
+            - dimensions: 300
+            - centroids: 20
+        2. Get and check the following property values:
+            - encoding: 8-Bit Scalar Quantizer Encoding
+            - metric: Euclidean Distance
+            - minTrainingSize: 25 * centroids
+            - maxTrainingSize: 256 * centroids
+        3. To check the encoding type, platform code will have to expose some internal property to the tests for verification.
+ */
+    func testVectorIndexConfigurationDefaultValue() {
+        let config = try! VectorIndexConfiguration(expression: "vector", dimensions: 300, centroids: 20)
```

Comment:
VectorIndexConfiguration has no `throws` so `try !` is not needed. Same for the other tests.